.github/workflows/ci.yml

```diff
@@ -98,63 +98,49 @@ jobs:
       - run: ./scalafmt --test
       - run: yarn install
       - run: yarn format-check
-  native-image:
+  scala-native:
     permissions:
       contents: write  # for actions/upload-release-asset to upload release asset
-    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [macOS-latest, ubuntu-latest]
-        libc: [default, musl]
-        exclude:
-          - os: macOS-latest
-            libc: musl
-        include:
-          - os: macOS-latest
-            artifact: scalafmt-macos
-            env:
-              NATIVE_IMAGE_STATIC: none
-          - os: ubuntu-latest
-            libc: default
-            artifact: scalafmt-linux-glibc
-            env:
-              NATIVE_IMAGE_STATIC: nolibc
-          - os: ubuntu-latest
-            libc: musl
-            artifact: scalafmt-linux-musl
-            env:
-              NATIVE_IMAGE_STATIC: musl
-    env: ${{ matrix.env }}
+        deploy: [ 
+          { os : macOS-13, name: scalafmt-x86_64-apple-darwin, path: "scalafmt-cli/native/target/scala-2.13/scalafmt-cli"}, 
+          { os : macOS-14, name: scalafmt-aarch64-apple-darwin, path: "scalafmt-cli/native/target/scala-2.13/scalafmt-cli"}, 
+          { os : ubuntu-latest, name: scalafmt-x86_64-pc-linux, path: "scalafmt-cli/native/target/scala-2.13/scalafmt-cli"}, 
+          { os : ubuntu-24.04-arm, name: scalafmt-aarch64-pc-linux, path: "scalafmt-cli/native/target/scala-2.13/scalafmt-cli"}, 
+          { os : windows-latest, name: scalafmt-x86_64-pc-win32, path: "scalafmt-cli/native/target/scala-2.13/scalafmt-cli.exe"}
+        ]
+    runs-on: ${{ matrix.deploy.os }}
     steps:
       - uses: actions/checkout@v4
-      - name: Set up GraalVM
-        uses: graalvm/setup-graalvm@v1
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-java@v4
         with:
           java-version: '21'
-          distribution: 'graalvm-community'
-          native-image-musl: ${{ matrix.libc == 'musl' }}
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          distribution: 'temurin'
+          cache: 'sbt'
       - uses: sbt/setup-sbt@v1
-      - run: bin/build-native-image.sh
+      - run: sbt scala-native
```

Comment:
should we create a script as well, `bin/build-scala-native.sh` and hide all the overly specific logic pertaining to `scala-2.13` or target location or current architecture (what you have under deploy.XXX)?

Reply:
I can do that for sure, but it's not a lot of logic really and I would have to also deal with windows scripts 😅

Reply:
let me continue this comment above, under `deploy`.